### ISSUE TITLE
Returnning the logger level

### ISSUE THREAD

**Issue by YangYongZhi** [zaproxy/zaproxy#4275]

**Comment by psiinon** [comment]
ZAP uses log4j, so the logger level can be set on a per class basis.
But it would be good to be able to get and set the logger levels via the API (and UI for that matter)

**Comment by YangYongZhi** [comment]
Thank you first.
I mean that I wanner know which level the log4j.propertis is using.
Because sometime If you set it as DEBUG, the zaproxy is running slow perhaps there will be too many IO operations, in order to see this level on web UI, I think I need a api to get the logger level setting.

**Comment by psiinon** [comment]
Yeah, I understood. But theres not actually just one level, there can be a different level per class. If we add API calls for this we should cope with the general case.

**Comment by kingthorin** [comment]
Would we want a UI for this as well or just API calls and scripts?

(I'm asking because I won't want to implement API actions and then find out we should have named them differently if they're options in the UI.)

I personally don't think it's necessary to have in the UI, given it can be handled simply with a standalone script and the majority of users shouldn't need to change it. 

I guess if we add a UI in the future it could also be a dialog or different type of panel in the UI.

**Comment by psiinon** [comment]
I dont think we need a UI right now. It could be a panel in the Options, but def not a priority for me.

**Comment by kingthorin** [comment]
So I've currently [added views](https://github.com/zaproxy/zaproxy/pull/8470) `loggingSummary` (which is limited to the top level paros and zap packages, similar to what was done for the CLI stuff) and `loggingDetails` (which loops through all loggers and spits out a map/table).

Shall I add an action with a generic name like `setLoggingDetails` that accepts a package/classname and lets the user set a level?

**Comment by thc202** [comment]
IMO `setLoggingLevel` with `name` and `level` as parameters.

**Comment by kingthorin** [comment]
Shall I rename `loggingSummary` to getLogLvel then? Suggestions for the detailed view, maybe `getDetailedLogLevel`?

For consistency.
